### PR TITLE
Build Freud's next branch in CI tests

### PR DESCRIPTION
**Pull request by tommy-waltmann**
## Description

Freud will be making a version 3 release, which will require updates to the examples. To update those examples, I recently created the `next` branch in this repo to mirror freud's `next` branch. To unit test the updated examples, we need CI to build freud's next branch instead of freud's master branch.

This PR implements the changes necessary for that to occur.

## Checklist:

- [ ] I agree with the terms of the [**freud Contributor Agreement**](https://github.com/glotzerlab/freud-examples/blob/master/ContributorAgreement.md).
- [ ] My code follows the code style of this project.
- [ ] I have updated the [index notebook](https://github.com/glotzerlab/freud-examples/blob/master/index.ipynb) if needed.
- [ ] My name is on the [contributors list](https://github.com/glotzerlab/freud-examples/blob/master/contributors.txt).

### CODE REVIEW

**Comment by tommy-waltmann** [comment]
A note: The unit tests should fail on this branch if it is ready for merge, we have not updated the examples for the API changes introduced in freud version 3.

**Comment by tommy-waltmann** [comment]
After inspection of output, the tests appear to be failing on the right examples.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ on:
   # request source branch is in the same repository.
   push:
     branches:
-      - "master"
+      - "next"
```
Comment (joaander) [inline .github/workflows/test.yml:15]:
```suggestion
      - "master"
      - "next"
```
Run tests on both the master and next branches.

**Review by joaander — APPROVED** [review]
This change looks good. I have one suggestion for improvement.

**Pull request merged**